drivers/gpio/gpio_ite_it8xxx2.c

```diff
@@ -0,0 +1,424 @@
+	if ((trig & GPIO_INT_TRIG_LOW) && (trig & GPIO_INT_TRIG_HIGH)) {
+		both_tri_en = 1;
+	} else {
+		if (trig & GPIO_INT_TRIG_LOW) {
+			SET_MASK(*hl_trig, BIT(i));
+		} else if (trig & GPIO_INT_TRIG_HIGH) {
+			CLEAR_MASK(*hl_trig, BIT(i));
+		}
+	}
+	/* set wui , only gpiob gpiof, currently */
+	if (gpio_config->reg_addr == GPIO_GPDRB) {
+		SET_MASK(*(GPIO_GPDRB_wui[gpio_config->gpios_pin].reg_addr),
+			 GPIO_GPDRB_wui[gpio_config->gpios_pin].pin);
+		SET_MASK(*(GPIO_GPDRB_wui[gpio_config->gpios_pin].clear_addr),
+			 GPIO_GPDRB_wui[gpio_config->gpios_pin].pin);
+		if (both_tri_en == 1) {
+			SET_MASK(
+			*(GPIO_GPDRB_wui[gpio_config->gpios_pin].reg_addr),
+			GPIO_GPDRB_wui[gpio_config->gpios_pin].pin);
+		}
+	}
+	if (gpio_config->reg_addr == GPIO_GPDRF) {
+		SET_MASK(*(GPIO_GPDRF_wui[gpio_config->gpios_pin].reg_addr),
+			 GPIO_GPDRF_wui[gpio_config->gpios_pin].pin);
+		SET_MASK(*(GPIO_GPDRF_wui[gpio_config->gpios_pin].clear_addr),
+			 GPIO_GPDRF_wui[gpio_config->gpios_pin].pin);
+		if (both_tri_en == 1) {
+			SET_MASK(
+			*(GPIO_GPDRF_wui[gpio_config->gpios_pin].reg_addr),
+			GPIO_GPDRF_wui[gpio_config->gpios_pin].pin);
+		}
+	}
+	/* Enable IRQ */
+	ret = irq_connect_dynamic(
+		gpio_config->gpio_irq, 0, gpio_ite_isr, dev, 0);
+	ite_intc_irq_enable(gpio_config->gpio_irq);
+	return ret;
+}
+
+static const struct gpio_driver_api gpio_ite_driver_api = {
+	.pin_configure = gpio_ite_configure,
+	.port_get_raw = gpio_ite_port_get_raw,
+	.port_set_masked_raw = gpio_ite_port_set_masked_raw,
+	.port_set_bits_raw = gpio_ite_port_set_bits_raw,
+	.port_clear_bits_raw = gpio_ite_port_clear_bits_raw,
+	.port_toggle_bits = gpio_ite_port_toggle_bits,
+	.pin_interrupt_configure = gpio_ite_pin_interrupt_configure,
+};
+
+static int gpio_ite_init(struct device *dev)
+{
+	const struct gpio_ite_cfg *gpio_config = DEV_GPIO_CFG(dev);
+	uint8_t regv;
+	unsigned int gpcr_offset = 0x10;
+	uint32_t gpcr_reg;
+	uint32_t gpcr_reg_addr;
+
+	if (gpio_config->reg_addr == GPIO_GPDRM) {
+		gpcr_offset = 0x40;
+	}
+
+	gpcr_reg = ((gpio_config->reg_addr & 0xff)  - 1) * 8 + gpcr_offset;
+	gpcr_reg_addr = gpcr_reg | (gpio_config->reg_addr & 0xffffff00);
+
+	if (gpio_config->port_is_output) {
+		ite_write(gpcr_reg_addr + gpio_config->gpios_pin, 1, 0x40);
+		if (gpio_config->flag & GPIO_OUTPUT_INIT_HIGH) {
+			set_bit(gpio_config, gpio_config->gpios_pin, GPIO_HIGH);
+		} else {
+			set_bit(gpio_config, gpio_config->gpios_pin, GPIO_LOW);
+		}
+	} else {
+		ite_write(gpcr_reg_addr + gpio_config->gpios_pin, 1, 0x80);
+		regv = ite_read(gpio_config->reg_addr, gpio_config->reg_size);
+	}
+	if (gpio_config->flag & GPIO_INT_ENABLE) {
+		gpio_pin_interrupt_configure(dev,
+					     gpio_config->gpios_pin,
+					     gpio_config->flag);
+	}
+	return 0;
+}
+
+#define GPIO_ITE_DEV_CFG_DATA(n, idx, is_out) \
+static struct gpio_ite_data gpio_ite_data_##n##_##idx; \
+static const struct gpio_ite_cfg gpio_ite_cfg_##n##_##idx = {		\
+	.reg_addr = \
+				DT_REG_ADDR_BY_IDX(DT_PHANDLE_BY_IDX\
```

Comment:
nit: the formatting seems off here. The indentation doesn't look right.

(This doesn't block PR, but if you are fixing other things, we can probably fix this too)